unity-renderer/Assets/UIComponents/Scripts/Components/SectionSelector/SectionToggle.cs

```diff
@@ -0,0 +1,93 @@
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+using static UnityEngine.UI.Toggle;
+
+public interface ISectionToggle
+{
+    /// <summary>
+    /// Event that will be triggered when the toggle is selected.
+    /// </summary>
+    ToggleEvent onSelect { get; set; }
+
+    /// <summary>
+    /// Get the toggle info.
+    /// </summary>
+    /// <returns>Model with all the toggle info.</returns>
+    SectionToggleModel GetInfo();
+
+    /// <summary>
+    /// Set the toggle info.
+    /// </summary>
+    /// <param name="model">Model with all the toggle info.</param>
+    void SetInfo(SectionToggleModel model);
+
+    /// <summary>
+    /// Invoke the action of selecting the toggle.
+    /// </summary>
+    void SelectToggle();
+}
+
+public class SectionToggle : MonoBehaviour, ISectionToggle
+{
+    [SerializeField] private Toggle toggle;
+    [SerializeField] private TMP_Text sectionText;
+    [SerializeField] private Image sectionImage;
+
+    public ToggleEvent onSelect
+    {
+        get { return toggle?.onValueChanged; }
+        set
+        {
+            toggle?.onValueChanged.RemoveAllListeners();
+            toggle?.onValueChanged.AddListener((isOn) =>
+            {
+                if (isOn)
+                    value.Invoke(isOn);
+            });
+        }
+    }
+
+    delegate void Logger(string str);
```

Comment:
```suggestion
```
I think this is not used

Reply:
I don't have idea why this is there! xD Good catch, thanks! Done in: https://github.com/decentraland/unity-renderer/pull/1175/commits/01619b09be6ddedeb365182383418f3c0baf53f9